app/controllers/api/databases.php

```diff
@@ -1859,21 +1871,31 @@ function updateAttribute(
     ->param('key', '', new Key(), 'Attribute Key.')
     ->param('required', null, new Boolean(), 'Is attribute required?')
     ->param('default', null, new Nullable(new Text(0, 0)), 'Default value for attribute when not provided. Cannot be set when attribute is required.')
+    ->param('size', null, new Integer(), 'Maximum size of the string attribute.', true)
     ->inject('response')
     ->inject('dbForProject')
     ->inject('queueForEvents')
-    ->action(function (string $databaseId, string $collectionId, string $key, ?bool $required, ?string $default, Response $response, Database $dbForProject, Event $queueForEvents) {
+    ->action(function (string $databaseId, string $collectionId, string $key, ?bool $required, ?string $default, ?int $size, Response $response, Database $dbForProject, Event $queueForEvents) {
 
-        $attribute = updateAttribute(
-            databaseId: $databaseId,
-            collectionId: $collectionId,
-            key: $key,
-            dbForProject: $dbForProject,
-            queueForEvents: $queueForEvents,
-            type: Database::VAR_STRING,
-            default: $default,
-            required: $required
-        );
+        try {
+            $attribute = updateAttribute(
+                databaseId: $databaseId,
+                collectionId: $collectionId,
+                key: $key,
+                dbForProject: $dbForProject,
+                queueForEvents: $queueForEvents,
+                type: Database::VAR_STRING,
+                default: $default,
+                required: $required,
+                size: $size
+            );
+        } catch (DatabaseException $e) {
+            if ($e->getMessage() === "Resize would result in data truncation") {
+                throw new Exception(Exception::ATTRIBUTE_INVALID_RESIZE);
+            } else {
+                throw $e;
+            }
+        }
```

Comment:
Let's add the try/catch directly around the `$dbForProject->updateAttribute` call inside the `updateAttribute` function to keep the scope narrow